### PR TITLE
feature/mx-1649 search preview merged

### PR DESCRIPTION
**Pull request by cutoffthetop**
# PR Context
- needs https://github.com/robert-koch-institut/mex-common/pull/347
- prep for https://github.com/robert-koch-institut/mex-editor/pull/192

# Added
- allow item merging functions to ignore cardinality and output preview items

# Changes
- create new endpoint for fetching previews of merged items
- replaced `mex.backend.fields` with `mex.common` counterpart

# Removed
- removed over-engineered `reraising` function

### CODE REVIEW

File: mex/backend/preview/main.py [hunk 1]
```diff
@@ -15,8 +20,34 @@ def preview_item(
     ruleSet: AnyRuleSetRequest,
 ) -> AnyMergedModel:
     """Preview the merging result when the given rule would be applied."""
+    # TODO(ND): Allow this endpoint to return "broken" merged items as well
```
Comment (rababerladuseladim) [inline mex/backend/preview/main.py:23]:
```suggestion
    # TODO(ND): Allow this endpoint to return merged items that fail cardinality validation.
```
Did you create a ticket for this?

File: mex/backend/settings.py [hunk 2]
```diff
@@ -71,7 +71,8 @@ class BackendSettings(BaseSettings):
         validation_alias="MEX_BACKEND_API_USER_DATABASE",
     )
     identity_provider: IdentityProvider | BackendIdentityProvider = Field(
-        IdentityProvider.MEMORY,
+        # IdentityProvider.MEMORY,
```
Comment (rababerladuseladim) [inline mex/backend/settings.py:74]: Why did the default change?

File: pyproject.toml [hunk 3]
```diff
@@ -11,7 +11,7 @@ dependencies = [
     "fastapi>=0.115,<1",
     "httpx>=0.27,<1",
     "jinja2>=3,<4",
-    "mex-common @ git+https://github.com/robert-koch-institut/mex-common.git@0.42.0",
+    "mex-common @ git+https://github.com/robert-koch-institut/mex-common.git@feature/mx-1649-add-preview-models",
```
Comment (rababerladuseladim) [inline pyproject.toml:14]: please update to a tagged version prior to merging.

File: mex/backend/merged/helpers.py [hunk 4]
```diff
@@ -126,26 +161,43 @@ def create_merged_item(
...
-        ValidationError,
-        InconsistentGraphError,
-        cls.model_validate,
-        merged_dict,
-    )
-    return cast(AnyMergedModel, merged_item)  # mypy, get a grip!
+    try:
+        return cls.model_validate(merged_dict)
+    except ValidationError as error:
+        raise InconsistentGraphError from error
```
Comment (rababerladuseladim) [inline mex/backend/merged/helpers.py:167]: This is not necessarily true as the data leading to the broken merged item could have been provided via API. Make it a `MExError` instead?

File: tests/merged/test_helpers.py [hunk 5]
```diff
@@ -225,28 +231,52 @@ def test_apply_subtractive_rule() -> None:
...
+                "identifier": Identifier.generate(seed=42),
+                "responsibleUnit": [Identifier.generate(seed=98)],
+            },
+        ),
     ],
     ids=(
-        "extracted_items_and_rule_set",
-        "only_rule_set",
-        "only_extracted_items",
+        "extracted items and rule set",
```
Comment (rababerladuseladim) [inline tests/merged/test_helpers.py:257]: These cases do neither check the type of the returned item (merged or preview) nor test if validation fails if the `validate_cardinality` is set and the provided params lead to a broken merged item. Could you expand the test cases?

**Review by rababerladuseladim — CHANGES_REQUESTED** [review]
Nice!

I only have some minor remarks.

File: mex/backend/preview/main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (cutoffthetop) [reply mex/backend/preview/main.py]: done 2392297

Comment (cutoffthetop) [reply mex/backend/preview/main.py]: MX-1759

File: mex/backend/settings.py (hunk 2 above)
Comment (cutoffthetop) [reply mex/backend/settings.py]: it shouldn't have. fixed in f5cd8ca

File: tests/merged/test_helpers.py (hunk 5 above)
Comment (cutoffthetop) [reply tests/merged/test_helpers.py:261]: > These cases do neither check the type of the returned item 

fixed 1f2091f

Comment (cutoffthetop) [reply tests/merged/test_helpers.py:300]: > Could you expand the test cases?

see cf1c39d

**Review by rababerladuseladim — APPROVED** [review]

File: pyproject.toml (hunk 3 above)
Comment (cutoffthetop) [reply pyproject.toml]: 3a256ba

**Pull request merged**